miniapps/tribol/README.md

```diff
@@ -0,0 +1,104 @@
+enabling the research and development of advanced contact algorithms. More
+information about Tribol can be found on [its Github
+repo](https://github.com/LLNL/Tribol).
+
+Tribol uses MFEM's data structures to simplify interoperability with MFEM-based
+finite element codes.  As a result, Tribol has a dependency on MFEM.  Other than
+this miniapp, MFEM has no dependencies on Tribol.  While a single MFEM install
+can be used to build and link both Tribol and this miniapp, it is generally
+simpler to allow Tribol's build system to build its own MFEM, then build the
+miniapp in a separately cloned repo. The installation instructions below follow
+this procedure.
+
+This directory contains an example using Tribol's mortar method to solve a
+contact patch test. A contact patch test places two aligned linear elastic cubes
+in contact, then verifies the exact elasticity solution for this problem is
+recovered. The exact solution requires transmission of a uniform pressure field
+across a non-conforming interface, which Tribol's mortar implementation is able
+to do. The test assumes small deformations and no accelerations so relationship
+between forces/contact pressures and deformations/contact gaps is linear and,
+therefore, the problem can be solved exactly with a single linear solve. The
+mortar implementation is based on [Puso and Laursen
+(2004)](https://doi.org/10.1016/j.cma.2003.10.010).  A description of the Tribol
+implementation is available in [Serac
+documentation](https://serac.readthedocs.io/en/latest/sphinx/theory_reference/solid.html#contact-mechanics).
+Lagrange multipliers are used to solve for the pressure required to prevent
+violation of the contact constraints.
+
+## Installation
+
+Tribol has dependencies on [Axom](https://github.com/LLNL/axom) and MFEM.  The
+simplest way to satisfy these dependencies is to follow the build instructions
+provided in the Tribol repo. This will use uberenv and spack to build
+dependencies (and their dependencies) mostly automatically. To simplify Axom's
+dependencies, the following spack spec is recommended:
+`^axom~shared~examples~tools~raja~umpire`. Furthermore, the version of MFEM
+built by spack should match the version of MFEM you are using. For instance, if
+you are tracking the latest `master` branch, the following spack spec is
+recommended: `^mfem@develop`. It's important to keep the version of MFEM Tribol
+uses in sync with the version the Tribol miniapp is using. If the MFEM version
+falls out of sync, the code will likely still compile, but will not run
+correctly. The easiest way to ensure the version of MFEM used by both Tribol and
+the miniapp are in sync are to use the same MFEM library for both, though this
+is outside the scope of this guide. After building Tribol, issue the command
+`make install` to simplify building and linking in MFEM.
+
+For the Tribol MFEM miniapps to successfully build and link, the MFEM build must
```

Comment:
After I built Tribol on quartz, axom was located in `<TRIBOL_ROOT>/../tribol_libs/<COMPILER>/axom-<AXOM_VER>-<AXOM_HASH>`. Similarly for conduit.